arch/risc-v/src/common/espressif/Make.defs

```diff
@@ -146,7 +146,14 @@ endif
 
 chip/$(ESP_HAL_3RDPARTY_REPO):
 	$(Q) echo "Cloning Espressif HAL for 3rd Party Platforms"
-	$(Q) git clone --quiet $(ESP_HAL_3RDPARTY_URL) chip/$(ESP_HAL_3RDPARTY_REPO)
+	$(Q) if [ -d "$(NXTMPDIR)" ]; then \
+			if [ ! -d "$(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO)" ]; then \
+				git clone --quiet $(ESP_HAL_3RDPARTY_URL) $(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO); \
+			fi; \
+			cp -fr $(NXTMPDIR)/$(ESP_HAL_3RDPARTY_REPO) chip/$(ESP_HAL_3RDPARTY_REPO); \
+		 else \
```

Comment:
should we add a download function to avoid the code dup? cmake version need similar change too.

Reply:
@xiaoxiang781216 Yes, I think it needs to be optimized and later for cmake as well.